lib/minify.js

```diff
@@ -291,11 +288,21 @@ async function minify(files, options, _fs_module) {
         toplevel.print(stream);
         result.code = stream.get();
         if (options.sourceMap) {
-            if(options.sourceMap.asObject) {
-                result.map = options.format.source_map.get().toJSON();
-            } else {
-                result.map = options.format.source_map.toString();
-            }
+            Object.defineProperty(result, "map", {
+                configurable: true,
+                enumerable: true,
+                get() {
+                    const map = options.format.source_map.getEncoded();
+                    return (result.map = options.sourceMap.asObject ? map : JSON.stringify(map));
+                },
+                set(value) {
+                    Object.defineProperty(result, "map", {
+                        value,
+                        writable: true,
```

Comment:
```suggestion
                        writable: true,
                        enumerable: true,
```

Maybe I'm wrong but if someone writes:

```
const res = await minify(...)
res.map // trigger getter
return JSON.stringify(res)
```

Then .map wouldn't be in the JSON because it's no longer enumerable? I might be wrong as I don't use defineProperty a lot though.

Reply:
The property will inherit the enumerability/configurability of the current descriptor:
```js
const o = {};
Object.defineProperty(o, 'test', { get() {}, configurable: true, enumerable: true });
Object.defineProperty(o, 'test', { value: 1, writable: true });
Object.getOwnPropertyDescriptor(o, 'test');
// { value: 1, writable: true, enumerable: true, configurable: true }
```

The spec has:
1. https://tc39.es/ecma262/multipage/ordinary-and-exotic-objects-behaviours.html#sec-ordinarydefineownproperty, which calls
2. https://tc39.es/ecma262/multipage/ordinary-and-exotic-objects-behaviours.html#sec-validateandapplypropertydescriptor, which has

```
b. Else if IsAccessorDescriptor(current) is true and IsDataDescriptor(Desc) is true, then
  i. If Desc has a [[Configurable]] field, let configurable be Desc.[[Configurable]]; else let configurable be current.[[Configurable]].
  ii. If Desc has a [[Enumerable]] field, let enumerable be Desc.[[Enumerable]]; else let enumerable be current.[[Enumerable]].
```

Reply:
Good to know, thanks!